tests/kernel/sched/schedule_api/src/main.c

```diff
@@ -15,7 +15,8 @@ K_THREAD_STACK_EXTERN(ustack);
 
 void spin_for_ms(int ms)
 {
-#if defined(CONFIG_X86_64) && defined(CONFIG_QEMU_TARGET)
+#if (defined(CONFIG_SOC_SERIES_MPS2) || defined(CONFIG_X86_64)) \
```

Comment:
Apparently this was not needed for mps2_an385 with Qemu 3.1, but I have not been able to make the test pass with Qemu 3.1 for mps2 targets.

Reply:
I would like @andyross to comment here, whether forcing the workaround for MPS2 targets as well (as we already do for X86_64) is "ok" to do.

I am stressing again that this did not seem to be needed, earlier, with Qemu 3.1.

Reply:
Is it the same root cause? (dropping irqs when spinning on a counter)
Real hardware or just QEMU?
Think the comment below should be updated to detail why this is needed for ARMV8M

Reply:
>Real hardware or just QEMU?
I don't have real MPS2 hardware to test this. @galak who can help with this?

>Is it the same root cause? (dropping irqs when spinning on a counter)
I can't really be sure. All I can say is that when this work-around is not enabled, I get ASSERT in line 63 of `test_slice_scheduling.c` (min time is less than expected)

> Think the comment below should be updated to detail why this is needed for ARMV8M

Yes, sure. But not only for ARMv8-M, @andrewboie; as  I mentioned, with Qemu 4.0, this fails, now, also for mps2_an385 target, which is a Cortex-M3.

Reply:
@andrewboie added a comment about the spinning on a counter for MPS2+